codebundles/k8s-tail-logs-dynamic/runbook.robot

```diff
@@ -0,0 +1,173 @@
+    [Tags]    resource    application    workload    logs    state    ${container_name}    ${workload_name}
+    ${logs}=    RW.CLI.Run Cli
+    ...    cmd=${KUBERNETES_DISTRIBUTION_BINARY} --context=${CONTEXT} -n ${NAMESPACE} logs -l ${LABELS} --tail=${MAX_LOG_LINES} --max-log-requests=10 --limit-bytes=${MAX_LOG_BYTES} --since=${LOGS_SINCE} --container=${CONTAINER_NAME}
+    ...    show_in_rwl_cheatsheet=true
+    ...    render_in_commandlist=true
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${kubeconfig}
+    RW.Core.Add Pre To Report    Workload Logs:\n\n${logs.stdout}
+    ${history}=    RW.CLI.Pop Shell History
+    RW.Core.Add Pre To Report    Commands Used: ${history}
+
+Tail `${CONTAINER_NAME}` Application Logs For Stacktraces
+    [Documentation]    Performs an inspection on container logs for exceptions/stacktraces, parsing them and attempts to find relevant source code information
+    [Tags]
+    ...    application
+    ...    debug
+    ...    app
+    ...    errors
+    ...    troubleshoot
+    ...    workload
+    ...    api
+    ...    logs
+    ...    ${container_name}
+    ...    ${workload_name}
+    ${cmd}=    Set Variable
+    ...    ${KUBERNETES_DISTRIBUTION_BINARY} --context=${CONTEXT} -n ${NAMESPACE} logs -l ${LABELS} --tail=${MAX_LOG_LINES} --max-log-requests=10 --limit-bytes=${MAX_LOG_BYTES} --since=${LOGS_SINCE} --container=${CONTAINER_NAME}
+    IF    $EXCLUDE_PATTERN != ""
+        ${cmd}=    Set Variable
+        ...    ${cmd} | grep -Ev "${EXCLUDE_PATTERN}" || true
+    END
+    ${logs}=    RW.CLI.Run Cli
+    ...    cmd=${cmd}
+    ...    show_in_rwl_cheatsheet=true
+    ...    render_in_commandlist=true
+    ...    env=${env}
+    ...    secret_file__kubeconfig=${kubeconfig}
+    ${parsed_stacktraces}=    RW.K8sApplications.Dynamic Parse Stacktraces    ${logs.stdout}
+    ...    parser_name=${STACKTRACE_PARSER}
+    ...    parse_mode=${INPUT_MODE}
+    ...    show_debug=True
+    ${report_data}=    RW.K8sApplications.Stacktrace Report Data   stacktraces=${parsed_stacktraces}
+    ${report}=    Set Variable    ${report_data["report"]}
+    ${mcst}=    Set Variable    ${report_data["most_common_stacktrace"]}
+    ${first_file}=    Set Variable    ${mcst.get_first_file_line_nums_as_str()}
+    ${history}=    RW.CLI.Pop Shell History
+    IF    (len($parsed_stacktraces)) > 0
+        RW.Core.Add Issue
+        ...    severity=3
+        ...    expected=No stacktraces were found in the application logs of ${CONTAINER_NAME}
+        ...    actual=Found stacktraces in the application logs of ${CONTAINER_NAME}
+        ...    reproduce_hint=Run:\n${cmd}\n view logs results for stacktraces.
+        ...    title=Stacktraces Found In Tailed Logs Of `${CONTAINER_NAME}`
+        ...    details=Generated a report of the stacktraces found to be reviewed.
+        ...    next_steps=Check this file ${first_file} for the most common stacktrace and review the full report for more details.
```

Comment:
are there any other valid next steps?  any other tasks that might be helpful?

Reply:
In transit right now - I have an experiment I'm working on to extend the next steps to be more specific based on what was parsed. Hoping to sneak it in tomorrow. For now this typically includes the line numbers and filename the exception originated from.